python/language.py

```diff
@@ -0,0 +1,48 @@
+from ruamel.yaml import YAML
+import os
+
+class LanguageExtractor():
+	def __init__(self):
+		self.lang_map = self.get_language_map()
+
+	def get_language_map(self):
+		yaml=YAML(typ='safe')
+		with open("../languages.yml") as f:
+			lang_map = f.read()
+		lang_map = yaml.load(lang_map)
+		return lang_map
+
+	def get_language_from_file(self, filename):
+		_, file_extension = os.path.splitext(filename)
+		fn_options = []
+		ext_options = []
+		for key in self.lang_map.keys():
```

Comment:
To optimize this further, we can create filename to language map and extensions to language at start and use that here.